### ISSUE TITLE
LoginActivity window has been leaked 

### ISSUE THREAD

**Issue by kunall17** [zulip/zulip-android#365]
```
02-04 01:26:14.564 2471-2471/com.zulip.android.dev E/WindowManager: android.view.WindowLeaked: Activity com.zulip.android.activities.LoginActivity has leaked window com.android.internal.policy.PhoneWindow$DecorView{706e3e6 V.E...... R......D 0,0-729,232} that was originally added here
                                                                        at android.view.ViewRootImpl.<init>(ViewRootImpl.java:368)
                                                                        at android.view.WindowManagerGlobal.addView(WindowManagerGlobal.java:299)
                                                                        at android.view.WindowManagerImpl.addView(WindowManagerImpl.java:85)
                                                                        at android.app.Dialog.show(Dialog.java:319)
                                                                        at com.zulip.android.activities.LoginActivity.onClick(LoginActivity.java:541)
                                                                        at java.lang.reflect.Method.invoke(Native Method)
                                                                        at android.support.v7.app.AppCompatViewInflater$DeclaredOnClickListener.onClick(AppCompatViewInflater.java:288)
                                                                        at android.view.View.performClick(View.java:5198)
                                                                        at android.view.View$PerformClick.run(View.java:21147)
                                                                        at android.os.Handler.handleCallback(Handler.java:739)
                                                                        at android.os.Handler.dispatchMessage(Handler.java:95)
                                                                        at android.os.Looper.loop(Looper.java:148)
                                                                        at android.app.ActivityThread.main(ActivityThread.java:5417)
                                                                        at java.lang.reflect.Method.invoke(Native Method)
                                                                        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
                                                                        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
```

On the branch https://github.com/sam1301/zulip-android/tree/use-server-ids

**Comment by harshitagupta30** [comment]
@kunall17 Can I work on this issue?

**Comment by kunall17** [comment]
@harshitagupta30 sure go ahead 👍

**Comment by harshitagupta30** [comment]
@kunall17 The leak in the LoginActivity is due to not closing the dialog box when we've exited the activity. The issue is in Sam1301's local branch and I couldn't reproduce it in master branch. But to avoid this issue we can call dailogbox.dismiss() method on Activity's onDestroy method.    
So can I make changes on master branch?

**Comment by kunall17** [comment]
> The issue is in Sam1301's local branch

There was no change in the LoginActivity nor in ZulipActivity, so it is not possible that this problem persist in her branch only! 
And i remember i got this in some other branch as well, send a PR for this I'll debug!

**Comment by Sam1301** [comment]
@kunall17 can this be closed? @harshitagupta30 thanks for submitting pr #391 that addresses this issue. I saw that you included "Issue#365" in the commit message. The way we prefer to add issue number in the commit messages is by including "Fixes #365" as the last line of commit messages(I guess you missed the space after "Issue"). That way issue is auto-closed once that commit gets merged. See the [commit message guideline](https://zulip.readthedocs.io/en/latest/version-control.html#commit-messages) for more details on this. 

Thanks!
